reconcile/utils/gitlab_api.py

```diff
@@ -828,3 +830,29 @@ def get_personal_access_tokens(self) -> list[PersonalAccessToken]:
             list[PersonalAccessToken],
             self.gl.personal_access_tokens.list(get_all=True),
         )
+
+    @staticmethod
+    def get_directory_contents(
+        project: Project,
+        ref: str | None = None,
+        path: str | None = None,
+    ) -> dict[str, bytes]:
+        """
+        Get the contents of a directory in a project.
+
+        :param project: The project to get the contents from.
+        :param ref: The commit SHA to download. A tag, branch reference, or SHA can be used. If not specified, defaults to the tip of the default branch.
+        :param path: The subpath of the repository to download. If an empty string, defaults to the whole repository.
+        :return: A dictionary with the file path as keys and the file content bytes as values.
+        """
+        archive = project.repository_archive(format="tar.gz", sha=ref, path=path)
```

Comment:
We need to keep an 👁️ on the memory usage. "Large" projects may exceed our current memory limits.

Reply:
they are all loaded in desired state, memory usage should be the same after `archive` GC